### ISSUE TITLE
Segment fault on Llama3 and Mixtral model using PyTorch/XLA nightly

### ISSUE THREAD

**Issue by zpcore** [pytorch/xla#8683]
We notice the segment fault issue when run llama3 and mixtral using v6e-256.

## 🐛 Bug

```
RAW: ExecuteFailureCallbacks() unsafe
RAW: Raising signal 11 with default behavior
bash: line 5:    10 Segmentation fault      python torchprime/launcher/thunk.py torchprime/torch_xla_models/train.py
```
Detailed gke log for internal:
[run link on 20250201](https://pantheon.corp.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22tpu-pytorch%22%0Aresource.labels.location%3D%22us-east1%22%0Aresource.labels.cluster_name%3D%22pytorch-v6e-256-donotdelete-uj%22%0Aresource.labels.namespace_name%3D%22default%22%0Aresource.labels.pod_name:%22piz-xpk-v6e-256-1-20250206-003239-slice-job-0-0-%22%20severity%3E%3DDEFAULT;storageScope=project;cursorTimestamp=2025-02-06T01:00:29.708952072Z;startTime=2025-02-06T00:32:33.909Z;endTime=2025-02-06T01:32:33.909Z?e=13802955&mods=allow_workbench_image_override&project=tpu-pytorch&inv=1&invt=AbozVA)
[run link on 20250131](https://pantheon.corp.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22tpu-pytorch%22%0Aresource.labels.location%3D%22us-east1%22%0Aresource.labels.cluster_name%3D%22pytorch-v6e-256-donotdelete-uj%22%0Aresource.labels.namespace_name%3D%22default%22%0Aresource.labels.pod_name:%22piz-xpk-v6e-256-1-20250206-024700-slice-job-0-0-%22%20severity%3E%3DDEFAULT;storageScope=project;cursorTimestamp=2025-02-06T02:48:50.601533828Z;duration=P1D?e=13802955&mods=allow_workbench_image_override&project=tpu-pytorch&inv=1&invt=Abo0Bg)

## To Reproduce

- Install https://github.com/AI-Hypercomputer/torchprime
- `tp use ...`
(Using dockerfile `us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/xla:nightly_3.10_tpuvm_cxx11_20250201`)  
- `tp run torchprime/torch_xla_models/train.py`

**Comment by zpcore** [comment]
[run link on 20250130 pass](https://pantheon.corp.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22tpu-pytorch%22%0Aresource.labels.location%3D%22us-east1%22%0Aresource.labels.cluster_name%3D%22pytorch-v6e-256-donotdelete-uj%22%0Aresource.labels.namespace_name%3D%22default%22%0Aresource.labels.pod_name:%22piz-xpk-v6e-256-1-20250206-035526-slice-job-0-0-%22%20severity%3E%3DDEFAULT;storageScope=project;cursorTimestamp=2025-02-06T04:24:49.724408805Z;duration=P1D?e=13802955&mods=allow_workbench_image_override&project=tpu-pytorch&inv=1&invt=Abo0OA)

**Comment by zpcore** [comment]
The issue should be related to the libtpu==0.0.8. 

20250201's nightly build is coupled with the libtpu==0.0.8. However,  when I try [run on 20250201 with libtpu==0.0.7](https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22tpu-pytorch%22%0Aresource.labels.location%3D%22us-east1%22%0Aresource.labels.cluster_name%3D%22pytorch-v6e-256-donotdelete-uj%22%0Aresource.labels.namespace_name%3D%22default%22%0Aresource.labels.pod_name:%22piz-xpk-v6e-256-1-20250206-041447-slice-job-0-0-%22%20severity%3E%3DDEFAULT;storageScope=project;duration=P1D?e=13802955&mods=allow_workbench_image_override&project=tpu-pytorch), the test can pass without segment fault.

Suggest wait for this week's libtpu pin update. I can file a bug to libtpu team if the issue still exists.

**Comment by tengyifei** [comment]
The pin updated just happened, updating us to `libtpu-nightly==0.0.10.dev20250210+nightly`. However, that build is no good either. It has the same crash when I tested: https://github.com/AI-Hypercomputer/torchprime/pull/113/files

I think we should file a bug to the libtpu team.

**Comment by zpcore** [comment]
There is a bug here b/395732855. Maybe we should make a libtpu debug build to see where the core dump actually happens.

**Comment by tengyifei** [comment]
*nod* if you could find sometime to make a run that would be much appreciated. we're stuck on the 2.6 version of torch_xla

**Comment by zpcore** [comment]
According to libtpu team, flag `--megascale_grpc_enable_xor_tracer=true` is the default in the failing libtpu-nightly, which caused the issue. 

We can either set XLA flag `--megascale_grpc_enable_xor_tracer=false` or wait under next libtpu nightly build (maybe 02/20/2025), which will default the flag to false.
